File: android/app/src/main/java/voidawt/image/ImageConsumer.java
package voidawt.image;

public interface ImageConsumer {
    int RANDOMPIXELORDER = 1;
    int TOPDOWNLEFTRIGHT = 2;
    int COMPLETESCANLINES = 4;
    int SINGLEPASS = 8;
    int SINGLEFRAME = 16;
    int IMAGEERROR = 1;
    int SINGLEFRAMEDONE = 2;
    int STATICIMAGEDONE = 3;
    int IMAGEABORTED = 4;

    void setDimensions(int width, int height);

    void setProperties(java.util.Hashtable<?, ?> props);

    void setColorModel(ColorModel model);

    void setHints(int hintflags);

    void setPixels(int x, int y, int w, int h, ColorModel model, byte[] pixels, int off, int scansize);

    void setPixels(int x, int y, int w, int h, ColorModel model, int[] pixels, int off, int scansize);

    void imageComplete(int status);
}

File: src/commonMain/kotlin/awt/image/ImageConsumer.kt
package awt.image

expect interface ImageConsumer {
    fun setDimensions(width: Int, height: Int)
    fun setColorModel(model: ColorModel?)
    fun setHints(hintflags: Int)
    fun setPixels(
        x: Int, y: Int, w: Int, h: Int,
        model: ColorModel, pixels: ByteArray, off: Int, scansize: Int,
    )
    fun setPixels(
        x: Int, y: Int, w: Int, h: Int,
        model: ColorModel?, pixels: IntArray?, off: Int, scansize: Int,
    )
    fun imageComplete(status: Int)
    fun setProperties(something: Any?)
}

// ImageConsumer's static flags.
expect val IMAGE_CONSUMER_RANDOMPIXELORDER: Int
expect val IMAGE_CONSUMER_TOPDOWNLEFTRIGHT: Int
expect val IMAGE_CONSUMER_COMPLETESCANLINES: Int
expect val IMAGE_CONSUMER_SINGLEPASS: Int
expect val IMAGE_CONSUMER_SINGLEFRAME: Int
expect val IMAGE_CONSUMER_IMAGEERROR: Int
expect val IMAGE_CONSUMER_SINGLEFRAMEDONE: Int
expect val IMAGE_CONSUMER_STATICIMAGEDONE: Int
expect val IMAGE_CONSUMER_IMAGEABORTED: Int